framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -109,23 +112,35 @@ public boolean isNothing() {
     /**
      * Converts a this range to a 32-bit integral range.
      *
-     * <p>If this range is too wide, i.e., wider than the full range of the Integer class, return
-     * INT_EVERYTHING.
+     * <p>If this range ignores overflow and the one of the bounds is outside the Integer range,
+     * then that bound is set to the bound of the Integer range.
+     *
+     * <p>If this range does not ignore overflow and is too wide, i.e., wider than the full range of
+     * the Integer class, return INT_EVERYTHING.
      *
-     * <p>If the bounds of this range are not representable as 32-bit integers, convert the bounds
-     * to Integer type in accordance with Java overflow rules, e.g., Integer.MAX_VALUE + 1 is
-     * converted to Integer.MIN_VALUE.
+     * <p>If this range does not ignore overflow and the bounds of this range are not representable
+     * as 32-bit integers, convert the bounds to Integer type in accordance with Java overflow
+     * rules, e.g., Integer.MAX_VALUE + 1 is converted to Integer.MIN_VALUE.
      */
     public Range intRange() {
-        if (this.isWiderThan(integerWidth)) {
-            return INT_EVERYTHING;
+        if (this.isNothing()) {
+            return NOTHING;
+        }
+        if (IGNORE_OVERFLOW) {
+            return new Range(
+                    from < Integer.MIN_VALUE ? Integer.MIN_VALUE : from,
```

Comment:
Could you just use the max function and the min function below?  I find this a little hard to read.